cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramStateWriter.java

```diff
@@ -44,42 +45,48 @@ void start(ProgramRunId programRunId, ProgramOptions programOptions, @Nullable S
    *
    * @param programRunId the id of the program run
    * @param twillRunId the run id of the twill application
+   * @param programOptions options passed to the program run
    */
-  void running(ProgramRunId programRunId, @Nullable String twillRunId);
+  void running(ProgramRunId programRunId, @Nullable String twillRunId, ProgramOptions programOptions);
```

Comment:
why are these changes needed? When would the program options ever change during a run?

Reply:
it would not change, but its currently only available in start method, which wont be available for methods such as running, etc as they are called from Twill AM container. thats why we pass programOptions to these methods.